src/EPR.Calculator.Service.Function/Services/ErrorReportService.cs

```diff
@@ -29,121 +30,38 @@ public ErrorReportService(IDbLoadingChunkerService<ErrorReport> errorReportChunk
             if (pomDetails == null) throw new ArgumentNullException(nameof(pomDetails));
             if (orgDetails == null) throw new ArgumentNullException(nameof(orgDetails));
 
-            var orgIds = new HashSet<int>(orgDetails.Select(o => o.OrganisationId).OfType<int>());
-
-            var orgToSubs = BuildOrgToSubsDictionary(orgDetails);
-
             var errorReports = new List<ErrorReport>();
 
-            foreach (var orgGroup in pomDetails.GroupBy(p => p.OrganisationId))
-            {
-                if (!orgGroup.Key.HasValue || !orgIds.Contains(orgGroup.Key.Value))
-                {
-                    // Org missing → add errors per unique org+sub
-                    errorReports.AddRange(CreateErrorsForMissingOrg(orgGroup, calculatorRunId, createdBy));
-                    continue;
-                }
-
-                var orgId = orgGroup.Key.Value;
-                orgToSubs.TryGetValue(orgId, out var knownSubsForOrg);
-                knownSubsForOrg ??= new HashSet<string>(StringComparer.OrdinalIgnoreCase);
-
-                var uniqueSubsInPoms = orgGroup
-                    .Select(p => p.SubsidaryId)
-                    .Where(s => !string.IsNullOrWhiteSpace(s))
-                    .Distinct(StringComparer.OrdinalIgnoreCase)
-                    .ToList();
-
-                var missingSubs = uniqueSubsInPoms
-                                .Where(sub => sub != null && !knownSubsForOrg.Contains(sub!))
-                                .ToList();
+            var orgIds = orgDetails.Select(o => (o.OrganisationId, o.SubsidaryId, o.SubmitterId)).ToHashSet();
+            var pomIds = pomDetails.Select(p => (p.OrganisationId, p.SubsidaryId, p.SubmitterId)).ToHashSet();
 
-                if (missingSubs.Any())
-                {
-                    errorReports.Add(CreateOrgLevelError(orgId, calculatorRunId, createdBy));
-                    var safeSubs = uniqueSubsInPoms
-                        .Where(s => !string.IsNullOrWhiteSpace(s))!
-                        .Select(s => s!) 
-                        .ToList();
-
-                    errorReports.AddRange(CreateSubsidiaryErrors(orgId, safeSubs, calculatorRunId, createdBy));
-                }
+            pomIds.SymmetricExceptWith(orgIds); 
```

Comment:
Should this be `ExceptWith` since orgIds without corresponding poms should be a different error?